hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/upgrade/SCMUpgradeFinalizer.java

```diff
@@ -74,33 +70,107 @@ public boolean preFinalizeUpgrade() throws IOException {
     msg += "\n  New pipelines creation will remain frozen until Upgrade " +
         "is finalized.";
 
-    storageContainerManager.preFinalizeUpgrade();
+    PipelineManager pipelineManager = scm.getPipelineManager();
+    /**
+     * Ask pipeline manager to not create any new pipelines. Pipeline
+     * creation will remain frozen until postFinalizeUpgrade().
+     */
+    pipelineManager.freezePipelineCreation();
+
+    /**
+     * Ask all the existing data nodes to close any open containers and
+     * destroy existing pipelines
+     */
+    waitForAllPipelinesToDestroy(pipelineManager);
+
+    /**
```

Comment:
I am not sure what this comment is for here... but it seems that some code was moved from here at one point, and now the comment itself is pointless...

Reply:
The comment makes sense, hence keeping it after converting to regular comment syntax.